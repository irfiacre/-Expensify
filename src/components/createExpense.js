import React from 'react';

const createExpense = ()=>(
    
    <div>
    This is the createExpense;
    </div>

); 

export default createExpense;    