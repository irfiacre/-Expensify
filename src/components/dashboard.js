import React from 'react';
import ExpenseList from './expenseList';
import ExpenseListFilter from './expenseListFilter';

const dashboard = ()=>(
    <div>
        <div className="contentContainer">
            <h2 className="dashboardTitle">Sort By</h2>
        </div>
        <div className='main'>
            <ExpenseListFilter />
            <ExpenseList />
        </div>
    </div>
);


export default dashboard;
