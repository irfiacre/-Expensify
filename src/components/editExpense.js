import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './expenseForm';
import { editExpense, removeExpense } from '../actions/expenses';


const updateExpense = (props)=>{
    return(
    <div>
        <div className="pageHeader">
          <div className="contentContainer">
            <h1 className="pageHeaderTitle">Edit Expense</h1>
          </div>
        </div>
        <div className="contentContainer">
        <ExpenseForm 
            expense = { props.expense }
            onSubmit={ (expense)=>{
                props.dispatch(editExpense(props.match.params.id,expense));
                props.history.push('/');
            } }
        />

        <button 
        className="button remove"
        onClick={()=>{ 
            props.dispatch(removeExpense( { id: props.match.params.id } )); 
            props.history.push('/');
           }}> 
            Remove 
        </button>
    </div>
    </div>
);
};

const mapStateToProps = (state, props)=>({ 
    expense: state.expenses.find(expense => expense.id === props.match.params.id ),
});

export default connect(mapStateToProps)(updateExpense);
