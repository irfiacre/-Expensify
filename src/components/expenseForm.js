import React from 'react';
import moment from 'moment';
import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

export default class ExpenseForm extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            description : props.expense ? props.expense.description : '',
            amount: props.expense ? ( props.expense.amount/100 ).toString() : '',
            note: props.expense ? props.expense.note : '',
            createdAt : props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        };
    };

    onDescriptionChange =(e)=>{
        const description = e.target.value;
        this.setState(()=>({ description }));
    };

    onAmountChange =(e)=>{
        const amount = e.target.value;
        if( !amount || amount.match(/^\d{1,}(\.\d{0,2})?$/) ){
            this.setState(()=>({ amount }));
        };
    };

    onAddNoteChange =(e)=>{
        const note = e.target.value;
        this.setState(()=>({ note }));
    };

    onDateChange = (createdAt)=>{
        if( createdAt ){
            this.setState(()=>({ createdAt }));
        };
    };

    onFocusChange = ( { focused } )=>{
        this.setState(()=>({ calendarFocused: focused }));
    };

    onFormSubmit = (e)=>{
        e.preventDefault();
        if(!this.state.description || !this.state.amount){
            this.setState(()=>({ error: 'Please provide both the description and amount.' }));
        } else {
            this.setState(()=>({ error: '' }));
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat((this.state.amount), 10) *100,
                note: this.state.note,
                createdAt: this.state.createdAt.valueOf(),
            });
        }
    };


    render(){
        return(
            <div>
                { this.state.error && <p>{ this.state.error }</p>  }

                <form onSubmit = { this.onFormSubmit }>
                <input type='text' 
                    placeholder='Description'
                    value = { this.state.description }
                    onChange= { this.onDescriptionChange }
                    autoFocus
                />
                <input type='text' 
                    value= { this.state.amount }
                    placeholder='Amount'
                    onChange = { this.onAmountChange }
                />

                <SingleDatePicker
                    date={ this.state.createdAt }
                    onDateChange = { this.onDateChange }
                    focused = { this.state.calendarFocused }
                    onFocusChange = { this.onFocusChange }
                    numberOfMonths = { 1 }
                    isOutsideRange = { ()=> false }
                />
                
                <textarea 
                    placeholder='Add a note for your expense(optional)'
                    onChange={ this.onAddNoteChange }
                ></textarea>
                    <button> Add Expense </button>
                </form>
            </div>
        )
    }
};
