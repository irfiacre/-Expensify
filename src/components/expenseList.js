import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './expenseListItem';
import filteredSortedExpenses from '../selectors/expenses';

const expensesList =(props)=>{
    return (
        <div>
            <h1> Expenses List: </h1>
            { props.expenses.map((expense)=>( <ExpenseListItem key={ expense.id }  { ...expense } /> ) )  }
        </div>
    );
};

const mapStateToProps = (state)=>({
    expenses: filteredSortedExpenses( state.expenses, state.filters ),
});


export default connect(mapStateToProps)(expensesList);
