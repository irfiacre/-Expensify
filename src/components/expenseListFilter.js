import React from 'react';
import { connect } from 'react-redux';
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
import { v4 as uuidv4 } from 'uuid';
import { setTextFilter,setSortByAmount, setSortByDate, setStartDate,setEndDate } from '../actions/filters';

class ExpenseListFilter extends React.Component{
    state = {
        calendarFocused: null,
    }

    onDatesChange = ({ startDate, endDate })=>{
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    }

    onFocusChange=( calendarFocused )=>{
        this.setState(()=>({ calendarFocused }));
    }
    onSortChange = (e)=>{ 
        if(e.target.value === 'date'){
            return this.props.dispatch(setSortByDate());
        } else if(e.target.value === 'amount'){
            return this.props.dispatch(setSortByAmount());
        }
     }
     onTextChange =(e)=>{
        this.props.dispatch(setTextFilter(e.target.value));
     }

    render(){
        return (
            <div className='contentContainer'>
                <div className="inputGroup">
                <div className="inputGroupItem">
                <input type='text'
                    className="textInput" 
                    value={ this.props.filters.text } 
                    placeholder='Text...' 
                    onChange={ this.onTextChange } 
                />
                </div>
                <div className="inputGroupItem">
                <select 
                    className="select"
                    value={ this.props.filters.sortBy }
                    onChange={ this.onSortChange }
                >
                    <option value='date' >Date</option>
        
                    <option value='amount'>Amount</option>
                </select>
                </div>
                <div className="inputGroupItem">
                < DateRangePicker 
                    startDate = { this.props.filters.startDate }
                    startDateId = { uuidv4() }
                    endDate = { this.props.filters.endDate }
                    endDateId = { uuidv4() }
                    onDatesChange ={ this.onDatesChange }
                    focusedInput = { this.state.calendarFocused }
                    onFocusChange = { this.onFocusChange }
                    showClearDates = { true }
                    numberOfMonths = { 1 }
                    isOutsideRange = { ()=> false }
                />
               </div>
             </div>
            </div>
        );
    }
}
// const expenseListFilter = (props)=>

const mapDispatchToProps=()=>{

};

const mapStateToProps = (state)=>({  filters: state.filters  });

export default  connect(mapStateToProps)(ExpenseListFilter);

