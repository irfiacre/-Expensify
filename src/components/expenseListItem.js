import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export default ( { id, description, amount, createdAt })=> (
        <Link className="listItem" to={ `/edit/${ id }` }>  
          <div>
              <h3 className="listItemTitle">{description}</h3>
              <span className="listItemSubtitle">{moment(createdAt).format('MMMM Do, YYYY')}</span>
           </div>

           <h3 className="listItemData">{numeral(amount / 100).format('$0,0.00')}</h3>

        </Link>
);
