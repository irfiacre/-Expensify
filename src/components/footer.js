import React from 'react';

const Footer =()=>(
    <footer className= 'footer'>
        <span className='footerContent'>
        Copyright Â© irfiacre 2020
        </span>
    </footer>
); 

export default Footer;

