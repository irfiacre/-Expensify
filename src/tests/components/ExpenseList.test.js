import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesList } from '../../components/expenseList';
import expensesTestData from '../fixtures/expensesData';

test('Should render expenses list correctly', ()=>{
    const wrapper = shallow( <ExpensesList expenses={ expensesTestData } /> );
    expect(wrapper).toMatchSnapshot();
});

test('Should render no expenses', ()=>{
    const wrapper = shallow( <ExpensesList expenses={[ ]} /> );
    expect(wrapper).toMatchSnapshot();
});
