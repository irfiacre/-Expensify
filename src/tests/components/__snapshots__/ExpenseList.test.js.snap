// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expenses list correctly 1`] = `
<div
  className="contentContainer"
>
  <div
    className="listHeader"
  >
    <div
      className="mobile"
    >
      Expenses
    </div>
    <div
      className="deskTop"
    >
      Expense
    </div>
    <div
      className="deskTop"
    >
      Amount
    </div>
  </div>
  <div
    className="listBody"
  >
    <Component
      amount={195}
      createdAt={0}
      description="Gum Expense"
      id="1"
      key="1"
      note="Testing Expense 1"
    />
    <Component
      amount={109500}
      createdAt={-345600000}
      description="Rent Expense"
      id="2"
      key="2"
      note="Testing Expense 2"
    />
    <Component
      amount={4500}
      createdAt={345600000}
      description="Credit Card Expense"
      id="3"
      key="3"
      note="Testing Expense 3"
    />
    ;
  </div>
</div>
`;

exports[`Should render no expenses 1`] = `
<div
  className="contentContainer"
>
  <div
    className="listHeader"
  >
    <div
      className="mobile"
    >
      Expenses
    </div>
    <div
      className="deskTop"
    >
      Expense
    </div>
    <div
      className="deskTop"
    >
      Amount
    </div>
  </div>
  <div
    className="listBody"
  >
    <div
      className="noListMessage"
    >
      <span>
        No expenses
      </span>
    </div>
    ;
  </div>
</div>
`;
